base/compiler/typelattice.jl

```diff
@@ -736,22 +736,15 @@ widenconst(::LimitedAccuracy) = error("unhandled LimitedAccuracy")
 # state management #
 ####################
 
-issubstate(lattice::AbstractLattice, a::VarState, b::VarState) =
-    ⊑(lattice, a.typ, b.typ) && a.undef <= b.undef
-
 function smerge(lattice::AbstractLattice, sa::Union{NotFound,VarState}, sb::Union{NotFound,VarState})
     sa === sb && return sa
     sa === NOT_FOUND && return sb
     sb === NOT_FOUND && return sa
-    issubstate(lattice, sa, sb) && return sb
```

Comment:
So we should always `tmerge` to encourage faster local absint convergence? I'm not sure if widening is required here to guarantee absint convergence.

Reply:
I forget exactly how we detect convergence. I thought schanged was some part of it? Otherwise presumably we could just have used Union here and kept the accuracy from that.